tests/components/pilight/test_init.py

```diff
@@ -66,351 +61,340 @@ def set_callback(self, function):
+        await hass.services.async_call(
+            pilight.DOMAIN,
+            pilight.SERVICE_NAME,
+            service_data=service_data1,
+            blocking=True,
+        )
+        await hass.services.async_call(
+            pilight.DOMAIN,
+            pilight.SERVICE_NAME,
+            service_data=service_data2,
+            blocking=True,
+        )
+        service_data1["protocol"] = [service_data1["protocol"]]
+        service_data2["protocol"] = [service_data2["protocol"]]
+
+        async_fire_time_changed(hass, dt_util.utcnow())
+        await hass.async_block_till_done()
+        error_log_call = mock_pilight_error.call_args_list[-1]
+        assert str(service_data1) in str(error_log_call)
+
+        new_time = dt_util.utcnow() + timedelta(seconds=5)
+        async_fire_time_changed(hass, new_time)
+        await hass.async_block_till_done()
+        error_log_call = mock_pilight_error.call_args_list[-1]
+        assert str(service_data2) in str(error_log_call)
+
+
+@patch("homeassistant.components.pilight._LOGGER.error")
+@patch("homeassistant.components.pilight._LOGGER", _LOGGER)
+@patch("pilight.pilight.Client", PilightDaemonSim)
+async def test_start_stop(mock_pilight_error, hass):
+    """Check correct startup and stop of pilight daemon."""
+    with assert_setup_component(4):
+        assert await async_setup_component(hass, pilight.DOMAIN, {pilight.DOMAIN: {}})
+
+        # Test startup
+        await hass.async_start()
+        await hass.async_block_till_done()
+
+        error_log_call = mock_pilight_error.call_args_list[-2]
+        assert "PilightDaemonSim callback" in str(error_log_call)
+        error_log_call = mock_pilight_error.call_args_list[-1]
+        assert "PilightDaemonSim start" in str(error_log_call)
+
+        # Test stop
+        with patch.object(hass.loop, "stop"):
+            await hass.async_stop()
+        error_log_call = mock_pilight_error.call_args_list[-1]
+        assert "PilightDaemonSim stop" in str(error_log_call)
+
+
+@patch("pilight.pilight.Client", PilightDaemonSim)
+@patch("homeassistant.core._LOGGER.debug")
+async def test_receive_code(mock_debug, hass):
+    """Check if code receiving via pilight daemon works."""
+    with assert_setup_component(4):
+        assert await async_setup_component(hass, pilight.DOMAIN, {pilight.DOMAIN: {}})
+
+        # Test startup
+        await hass.async_start()
+        await hass.async_block_till_done()
+
+        expected_message = dict(
+            {
+                "protocol": PilightDaemonSim.test_message["protocol"],
+                "uuid": PilightDaemonSim.test_message["uuid"],
+            },
+            **PilightDaemonSim.test_message["message"],
+        )
+        debug_log_call = mock_debug.call_args_list[-3]
+
+        # Check if all message parts are put on event bus
+        for key, value in expected_message.items():
+            assert str(key) in str(debug_log_call)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)